docs/solidity-by-example.rst

```diff
@@ -645,99 +645,71 @@ Safe Remote Purchase
 sign and verify signatures, and setup the payment channel.
 
 Creating and verifying signatures
 =================================
 
 Imagine Alice wants to send a quantity of Ether to Bob, i.e.
 Alice is the sender and the Bob is the recipient.
+.. TODO: What does the below mean?
 Alice only needs to send cryptographically signed messages off-chain
 (e.g. via email) to Bob and it will be very similar to writing checks.
 
-Signatures are used to authorize transactions,
-and they are a general tool that is available to
-smart contracts. Alice will build a simple
-smart contract that lets her transmit Ether, but
-in a unusual way, instead of calling a function herself
-to initiate a payment, she will let Bob
-do that, and therefore pay the transaction fee.
+Alice and Bob use signatures to authorise transactions, which are a feature of smart contracts.
+Alice will build a simple smart contract that lets her transmit Ether, but instead of calling a function herself
+to initiate a payment, she will let Bob do that, and therefore pay the transaction fee.
+
 The contract will work as follows:
 
     1. Alice deploys the ``ReceiverPays`` contract, attaching enough Ether to cover the payments that will be made.
-    2. Alice authorizes a payment by signing a message with their private key.
+    2. Alice authorises a payment by signing a message with their private key.
     3. Alice sends the cryptographically signed message to Bob. The message does not need to be kept secret
-       (you will understand it later), and the mechanism for sending it does not matter.
+       (explained later), and the mechanism for sending it does not matter.
     4. Bob claims their payment by presenting the signed message to the smart contract, it verifies the
        authenticity of the message and then releases the funds.
 
 Creating the signature
 ----------------------
 
-Alice does not need to interact with Ethereum network to
-sign the transaction, the proccess is completely offline.
-In this tutorial, we will sign messages in the browser
-using ``web3.js`` and ``MetaMask``.
-In particular, we will use the standard way described in `EIP-762 <https://github.com/ethereum/EIPs/pull/712>`_,
+Alice does not need to interact with the Ethereum network to sign the transaction, the process is completely offline.
+In this tutorial, we will sign messages in the browser using `web3.js <https://github.com/ethereum/web3.js>`_ and `MetaMask <https://metamask.io>`_, using the method described in `EIP-762 <https://github.com/ethereum/EIPs/pull/712>`_,
 as it provides a number of other security benefits.
 
 ::
 
-    /// Hashing first makes a few things easier
-    var hash = web3.sha3("message to sign");
-    web3.personal.sign(hash, web3.eth.defaultAccount, function () {...});
-
+    /// Hashing first makes things easier
+    var hash = web3.utils.sha3("message to sign");
+    web3.eth.personal.sign(hash, web3.eth.defaultAccount, function () {
+      console.log("Signed")
```

Comment:
Perhaps this should better also output the signature.

Reply:
@chriseth In the original there was nothing inside the function and I added this mostly to know if it had worked. As the example adds to it later anyway, I'm thinking maybe just leave the function empty for now?